scripts/stac/imagery/create_stac.py

```diff
@@ -127,48 +126,76 @@ def create_item(
             )
 
     item.update_datetime(start_datetime, end_datetime)
-    item.update_spatial(geometry, bbox)
+    item.update_spatial(*get_extents(gdalinfo_result or gdal_info(asset_path)))
     item.add_collection(collection_id)
 
     get_log().info("ImageryItem created", path=asset_path)
     return item
 
 
-def create_base_item(asset_path: str, gdal_version: str) -> ImageryItem:
+def create_base_item(asset_path: str, gdal_version: str, current_datetime: str, published_path: str | None) -> ImageryItem:
     """
     Args:
         asset_path: path of the visual asset (TIFF)
         gdal_version: GDAL version string
+        current_datetime: datetime string that represents the current time when the item is created
+        published_path: path of the published dataset
 
     Returns:
         An ImageryItem with basic information.
     """
-    id_ = get_file_name_from_path(asset_path)
-    file_content = fs.read(asset_path)
-    file_modified_datetime = format_rfc_3339_datetime_string(modified(asset_path))
-
-    stac_asset = STACAsset(
-        **{
-            "href": os.path.join(".", os.path.basename(asset_path)),
-            "file:checksum": checksum.multihash_as_hex(file_content),
-            "created": file_modified_datetime,
-            "updated": file_modified_datetime,
-        }
-    )
 
-    now_string = format_rfc_3339_datetime_string(utc_now())
-
-    if (topo_imagery_hash := os.environ.get("GIT_HASH")) is not None:
+    if (topo_imagery_hash := environ.get("GIT_HASH")) is not None:
         commit_url = f"https://github.com/linz/topo-imagery/commit/{topo_imagery_hash}"
     else:
         commit_url = "GIT_HASH not specified"
 
     stac_processing = STACProcessing(
         **{
-            "processing:datetime": now_string,
+            "processing:datetime": current_datetime,
             "processing:software": STACProcessingSoftware(**{"gdal": gdal_version, "linz/topo-imagery": commit_url}),
-            "processing:version": os.environ.get("GIT_VERSION", "GIT_VERSION not specified"),
+            "processing:version": environ.get("GIT_VERSION", "GIT_VERSION not specified"),
+        }
+    )
+
+    id_ = get_file_name_from_path(asset_path)
+    file_content = fs.read(asset_path)
+    created_datetime = updated_datetime = current_datetime
+
+    if published_path:
+        try:
+            existing_item_content = read(path.join(published_path, f"{id_}.json"))
+            existing_item = json.loads(existing_item_content.decode("UTF-8"))
+            try:
+                created_datetime = existing_item["properties"]["created"]
+            except KeyError:
+                get_log().info(f"Existing Item {id_} does not have 'properties.created' attribute")
```

Comment:
In this case we should not set the created datetime at all - it needs to be backfilled.